lib/asn1/gen.c

```diff
@@ -797,8 +807,12 @@ define_type (int level, const char *name, const char *basename, Type *t, int typ
 		n = NULL;
 		pos++;
 	    }
-	    /* pad to 32 elements */
-	    while (pos < 32) {
+	    /* pad unused tail */
+            if (max_memno > 31)
+                max_memno += 64 - (max_memno % 64);
+            else
+                max_memno = 32;
```

Comment:
A brief comment explain why this is exactly right and not off by one could be cool.